### PR TITLE
Check addresses supplied by users for conflict with existing routes

### LINKED ISSUE #437: [dns] subnet collision hard to detect
A user [accidentally launched weavedns with a CIDR that is in use](https://botbot.me/freenode/weavenetwork/2015-03-06/?msg=33505887); in fact the host's nameserver lives on that sub-net. There is no indication of anything being amiss in the weavedns logs - just failed lookups for external domains.

We could detect this, i.e. check whether the entry in resolv.conf clashes with our weave CIDR.

Admittedly this is a very special case. There are plenty of other opportunities for subnet collisions. However, unlike the others, this one is easy to detect.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #437, #480.

Issues a warning but carries on.

Note that we are still not checking ranges passed to IPAM; I'm still considering whether that is a good idea.

### CODE REVIEW

File: net/route.go [hunk 1]
```diff
@@ -26,3 +26,18 @@ func CheckNetworkFree(subnet *net.IPNet) error {
...
+// For a specific address, we only care if it is actually *inside* an
+// existing route, because weave-local traffic never hits IP routing.
+func CheckAddressOverlap(addr net.IP) error {
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_V4)
+	if err != nil {
+		return err
+	}
+	for _, route := range routes {
+		if route.Dst != nil && route.Dst.Contains(addr) {
+			return fmt.Errorf("Address %s would overlap with existing route %s", addr, route.Dst)
```
Comment (rade) [inline net/route.go]: s/would overlap/overlaps

...since we don't treat this as an error, and carry on.

**Comment by rade** [comment]
Where does the warning pop up when using the proxy? In the CLI? The proxy logs?

**Comment by bboreham** [comment]
> Where does the warning pop up when using the proxy? In the CLI? The proxy logs?

Nowhere, it seems.

**Comment by bboreham** [comment]
#1056 created for proxy issue

**Pull request merged**